### PR TITLE
redfishtool: init at 1.1.8

### LINKED ISSUE #259243: Package request: redfishtool
**Project description**

Redfishtool is a cli client for managing remote hardware using the Redfish protocol

**Metadata**

* homepage URL: https://github.com/DMTF/Redfishtool
* source URL: https://github.com/DMTF/Redfishtool
* license: Proprietary
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by jfvillablanca**
## Description of changes
Resolves #259243

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 260447` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>redfishtool</li>
    <li>redfishtool.dist</li>
  </ul>
</details>

File: pkgs/by-name/re/redfishtool/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [ requests python-dateutil ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A Python34 program that implements a command line tool for accessing the Redfish API.";
+    homepage = "https://github.com/DMTF/Redfishtool";
+    license = licenses.unfree;
```
Comment (nevivurn) [inline pkgs/by-name/re/redfishtool/package.nix:26]: Skimming the upstream repo, it seems like it is supposed to be BSD 3-clause, which agrees with what the [Debian](https://metadata.ftp-master.debian.org/changelogs/main/r/redfishtool/redfishtool_1.1.0-3_copyright) folks have on their repositories. Is this actually unfree?

File: pkgs/by-name/re/redfishtool/package.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "redfishtool";
+  version = "1.1.8";
+in
+python3.pkgs.buildPythonApplication {
+  inherit pname version;
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-X/G6osOHCBidKZG/Y2nmHadifDacJhjBIc7WYrUCPn8=";
```
Comment (nevivurn) [inline pkgs/by-name/re/redfishtool/package.nix:16]:
```suggestion
    hash = "sha256-X/G6osOHCBidKZG/Y2nmHadifDacJhjBIc7WYrUCPn8=";
```

File: pkgs/by-name/re/redfishtool/package.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [ requests python-dateutil ];
+
+  doCheck = false;
```
Comment (nevivurn) [inline pkgs/by-name/re/redfishtool/package.nix:21]: There should probably be a comment explaining the reason why checks are disabled; the package seems to build fine with them enabled, too. Emits a few warnings, though.

**Review by nevivurn — COMMENTED** [review]
We should also set `meta.mainProgram`

File: pkgs/by-name/re/redfishtool/package.nix (hunk 1 above)
Comment (jfvillablanca) [reply pkgs/by-name/re/redfishtool/package.nix:26]: i was unsure when I read their license so I based the license on the Package Request which stated that it was proprietary. Will resolve all these issues. Thanks

File: pkgs/by-name/re/redfishtool/package.nix (hunk 3 above)
Comment (jfvillablanca) [reply pkgs/by-name/re/redfishtool/package.nix:21]: this was a mistake, i mistook this for another package where tests where flaky. will rectify

**Review by nevivurn — APPROVED** [review]

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 260447` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>redfishtool</li>
    <li>redfishtool.dist</li>
  </ul>
</details>

File: pkgs/by-name/re/redfishtool/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-X/G6osOHCBidKZG/Y2nmHadifDacJhjBIc7WYrUCPn8=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [ requests python-dateutil ];
+
+  meta = with lib; {
+    description = "A Python34 program that implements a command line tool for accessing the Redfish API.";
```
Comment (bbigras) [inline pkgs/by-name/re/redfishtool/package.nix:22]:
```suggestion
    description = "A Python34 program that implements a command line tool for accessing the Redfish API";
```

no trailing coma

**Review by nevivurn — APPROVED** [review]

**Pull request merged**